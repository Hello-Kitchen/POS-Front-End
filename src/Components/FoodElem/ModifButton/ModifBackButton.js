import { GoArrowDown } from "react-icons/go";
import { useNavigate } from "react-router-dom";

function ModifButton({id, food}) {

    const navigate = useNavigate();
    const handleClick = () => {
        navigate(-1, {state: {food: food}})
    }

    return (
        <div className="h-1/6 w-full grid grid-flow-col colbottom-0 p-6">
            <div className="w-full col-span-5 justify-items-center flex-row content-center">
                <h1 className="text-3xl font-bold text-black">
                    Modifications
                </h1>
            </div>
            <div className="col-span-1 w-full grid justify-items-end float-right">
                <button onClick={() => handleClick()}>
                    <GoArrowDown size={40} color="black" />
                </button>
            </div>
        </div>
    )
}

export default ModifButton;