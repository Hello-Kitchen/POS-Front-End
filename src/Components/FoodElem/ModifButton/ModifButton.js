import React from 'react';
import PropTypes from 'prop-types';

import { GoArrowRight } from "react-icons/go";

/**
 * Footer Component : used to change page from the detail modification page of a food to the ingredient modification page
 * 
 * @component ModifButton
 * @param {function} openModif function used to open the modification page
 */
function ModifButton({openModif}) {

    //Function called on button click, navigates to the food ingredients modification page
    const handleClick = () => {
        openModif();
    }

    return (
        <div className="w-full row-span-1 grid grid-flow-col colbottom-0 p-6 content-center border-t-2 border-t-black">
            <div className="w-full col-span-5 justify-items-start flex-row content-center">
                <h1 className="text-3xl font-bold text-black">
                    Modifications
                </h1>
            </div>
            <div className="col-span-1 w-full grid justify-items-end float-right">
                <button onClick={() => handleClick()}>
                    <GoArrowRight size={40} color="black" />
                </button>
            </div>
        </div>
    )
}


ModifButton.propTypes = {
    openModif: PropTypes.func.isRequired,
}

export default ModifButton;
