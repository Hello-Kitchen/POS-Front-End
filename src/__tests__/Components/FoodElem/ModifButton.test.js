import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import ModifButton from '../../../Components/FoodElem/ModifButton/ModifButton';

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: jest.fn(),
}));

describe('ModifButton Component', () => {
    const mockFood = { id: 1, name: 'Pizza', description: 'Delicious pizza' };
    const mockFoods = ['Burger Miam', 'Burger Gourmet'];
    const mockColor = 'red';

    test('renders', () => {
        const { getByText, getByRole } = render(
            <Router>
                <ModifButton food={mockFood} foods={mockFoods} color={mockColor} />
            </Router>
        );
      
        expect(getByText('Modifications')).toBeInTheDocument();
      
        const button = getByRole('button');
        expect(button).toBeInTheDocument();
      
        const icon = button.querySelector('svg');
        expect(icon).toBeInTheDocument();
    });

    test('matches the snapshot', () => {
        const { asFragment } = render(
            <Router>
                <ModifButton food={mockFood} foods={mockFoods} color={mockColor} />
            </Router>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
