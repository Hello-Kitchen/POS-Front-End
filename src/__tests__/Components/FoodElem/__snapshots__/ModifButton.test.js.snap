// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModifButton Component matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="w-full row-span-1 grid grid-flow-col colbottom-0 p-6 content-center border-t-2 border-t-black"
  >
    <div
      class="w-full col-span-5 justify-items-center flex-row content-center"
    >
      <h1
        class="text-3xl font-bold text-black"
      >
        Modifications
      </h1>
    </div>
    <div
      class="col-span-1 w-full grid justify-items-end float-right"
    >
      <button>
        <svg
          color="black"
          fill="currentColor"
          height="40"
          stroke="currentColor"
          stroke-width="0"
          style="color: black;"
          viewBox="0 0 24 24"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18.655 10.405a.75.75 0 0 1-1.06 0l-4.97-4.97v14.44a.75.75 0 0 1-1.5 0V5.435l-4.97 4.97a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734l6.25-6.25a.75.75 0 0 1 1.06 0l6.25 6.25a.75.75 0 0 1 0 1.06Z"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
